docs/picontrol.md

```diff
@@ -0,0 +1,77 @@
+
+## ID badges
+
+At TreeHacks 2020, we printed a custom badge for every hacker. We used [these pvc id cards](https://store.gototags.com/nfc-pvc-badge-vertical) and an ID card printer (TODO: which one? although any should work).
+
+The process we eventually used relied on the hardware IDs for each card being unique. This means that NTAG210-215 and MIFARE cards should work, but to be safe you should check the specification for the protocol you're planning on using to make sure that there won't be any ID collisions. From our research, it seems that most manufacturers like to assign IDs in sequence, but that's obviously not guaranteed.
+
+## Hardware
+
+Our hardware "cube" were assembled from the following:
+
+- A plastic cube case. These were surprisingly hard to find, so we bought [these](https://www.amazon.com/Mr-Go-Rechargeable-Soothing-Decorative-Nightstand/dp/B01E7685IG/ref=sr_1_2?dchild=1&keywords=Mr.%2BGO%2Blight%2Bcube&qid=1592967810&sr=8-2&th=1) and replaced their internals with our hardware.
+
+- Raspberry Pi Model 3b+ (but any Raspberry Pi with Wifi capability should work)
+
+- [USB NFC readers](https://www.amazon.com/gp/product/B07Q5KXM6J/ref=ppx_yo_dt_b_asin_title_o06_s00?ie=UTF8&psc=1). These emulate a keyboard -- when an NFC badge is scanned onto them, they 'type' its ID into whatever they're connected to.
+  - **PLEASE** make note of the model: if that link isn't dead, you'll see it links to a '14H' reader. '14H' (or '14D') is required to get the specificity necessary to distinguish between the NFC badge IDs: these readers read the most significant bit first, and since the card IDs are assigned sequentially this means that most of them have the same first 8-10 hex digits. You are in for much pain if you do not purchase the correct readers.
+- Microusb battery pack. Best to get flat ones.
+- *Optionally*: LEDs
+
+### Hardware: Future Considerations
+
+- Raspberry Pi's are certainly overkill for this application. Since the boxes only need to make a single web request, a simpler IoT board (eg. ESP8266 or ESP32) would work just as well, consume far less power, and cost an order of magnitude less.
+  - As a corollary, "PiControl" is probably a bad name for this system as pretty much all of the logic is hardware-agnostic. Even the small Python script run on the Pi could probably run on a much simpler microcontroller (with a couple modifications) through Micropython.
+- The keyboard emulation NFC readers were used because Katz (the author of this doc) had never touched hardware in his life and thus couldn't figure out how to use the [an actual breakout board](https://www.adafruit.com/product/364). In the first week of CS107e, he realized that he was connecting the board's TX and RX pins to the Pi's TX and RX pins (respectively). He has now learned that TX == "transmit" and RX == "receive" and thus TX on one board should be connected to RX on the other and vice versa. If you ever see Katz, feel free to chastise him for his foolishness.
```

Comment:
Not more efficient per se, but it would've saved us the hassle of requiring all IDs to be unique (and thus having to look around for readers that read the full ID) as we could've added our own IDs or other metadata onto the cards.

Also, this is something of a self-roast as I must've spent 10-15 hours debugging that. I thought it must've been an issue with my code / the dependencies and stupidly didn't consider that I had wired it up wrong.